projects/packages/videopress/src/class-ajax.php

```diff
@@ -205,8 +205,7 @@ public function wp_ajax_videopress_get_upload_token() {
 		);
 
 		$endpoint = "sites/{$video_blog_id}/media/token";
-		$result   = Client::wpcom_json_api_request_as_blog( $endpoint, Client::WPCOM_JSON_API_VERSION, $args );
-
+		$result   = Client::wpcom_json_api_request_as_blog( $endpoint, Client::WPCOM_JSON_API_VERSION, $args, array( 'external_user_id' => get_current_user_id() ) );
```

Comment:
I left a comment on D165318-code for why this `external_user_id` isn't always going to be enough to generate a user-specific token.

Reply:
Do we need to pass that `external_user_id` property? There's an already existing `Client::wpcom_json_api_request_as_user` method we can use to pass the user id.

Reply:
> Do we need to pass that external_user_id property? There's an already existing Client::wpcom_json_api_request_as_user method we can use to pass the user id.

For some reason when this function is used it fails with error `You must be logged-in to get an upload token` , I suspect because the upload endpoint is an older endpoint it might need some additional configuration for it to work. wanted to get eyes on the PR for discussion so didn't look into it too much. can revisit it.